### PR TITLE
`ObjectSelectorBase` add option to not throw on missing input

### LINKED ISSUE #45384: TkAlZMuMuGenMuonSelector: ProductNotFound (genParticles) in RelVals 1001.3, 1001.4, 1002.3, 1002.4
In 2024-07-04-2300 IBs, RelVals 1001.3, 1001.4, 1002.3 and 1002.4 failed with ProductNotFound exception:

```
Begin processing the 25th record. Run 355769, Event 22846252, LumiSection 35 on stream 3 at 05-Jul-2024 08:56:10.712 CEST
----- Begin Fatal Exception 05-Jul-2024 08:56:10 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 355769 lumi: 35 event: 22809306 stream: 1
   [1] Running path 'pathALCARECOTkAlZMuMu'
   [2] Calling method for module GenParticleSelector/'TkAlZMuMuGenMuonSelector'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::GenParticle>
Looking for module label: genParticles
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "TryToContinue = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

Probably caused by https://github.com/cms-sw/cmssw/pull/45357 - @mmusich FYI

### PR DESCRIPTION
**Pull request by mmusich**
fixes https://github.com/cms-sw/cmssw/issues/45384

#### PR description:

Title say it all, and use it to fix workflows broken as pointed out in https://github.com/cms-sw/cmssw/issues/45384#issue-2392129336

#### PR validation:

` runTheMatrix.py -l 1001.3` runs with this addition.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 1001.3, 1001.4, 1002.3, 1002.4

**Comment by mmusich** [comment]
type bug-fix

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CommonTools/UtilAlgos/interface/ObjectSelectorBase.h [hunk 1]
```diff
@@ -57,6 +58,9 @@ class ObjectSelectorBase : public Base {
   bool filter(edm::Event& evt, const edm::EventSetup& es) override {
     selectorInit_.init(selector_, evt, es);
     edm::Handle<typename Selector::collection> source;
+    if (!throwOnMissing_ && !source.isValid()) {
+      return false;
```
Comment (mmusich) [inline CommonTools/UtilAlgos/interface/ObjectSelectorBase.h:62]:
```suggestion
      return !filter_;
```

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by mmusich** [comment]
urgent
   * I believe this should remove failures from IB workflows.

**Comment by antoniovilela** [comment]
+1

**Pull request merged**